Templating/FilterTrait.php

```diff
@@ -42,7 +42,7 @@ public function __construct(CacheManager $cache)
      */
     public function filter($path, $filter, array $config = [], $resolver = null, $referenceType = UrlGeneratorInterface::ABSOLUTE_URL)
     {
-        return $this->cache->getBrowserPath(parse_url($path, PHP_URL_PATH), $filter, $config, $resolver, $referenceType);
+        return $this->cache->getBrowserPath($path, $filter, $config, $resolver, $referenceType);
```

Comment:
as the parse_url was added in b77558c98e389bef3a642f4a03745b47925b90d4 to remove query parameter, could we keep removing those but without losing the domain? otherwise we trade the fix for a regression in other cases, i think.

Reply:
I actually think we should not be removing the query parameters when getting the image. Indeed, some image provider will actually serve images based on them (could include size, path, etc). Nonetheless, parameters should be stripped at some points and this is why I also changed the WebPathResolver to take this into account. 

I am happy to revisit this if needed as I do believe that allowing handling remote files is a critical feature.

Reply:
> I actually think we should not be removing the query parameters when getting the image. Indeed, some image provider will actually serve images based on them (could include size, path, etc). Nonetheless, parameters should be stripped at some points and this is why I also changed the WebPathResolver to take this into account.
> 
> I am happy to revisit this if needed as I do believe that allowing handling remote files is a critical feature.

Agreed! And it's actually the conclusion of the discussion here :
https://github.com/liip/LiipImagineBundle/issues/1172#issuecomment-554983445

Reply:
why remove the query parameter? why not slug the whole string?

Reply:
@c33s I took your advice into account and change the method to have something more generic.